src/domain-functions.ts

```diff
@@ -49,40 +49,11 @@ function all<Fns extends DomainFunction[]>(
 function collect<Fns extends Record<string, DomainFunction>>(
   fns: Fns,
 ): DomainFunction<UnpackDFObject<Fns>> {
-  return async (input, environment) => {
-    const results = await Promise.all(
-      Object.entries(fns).map(
-        async ([key, fn]) =>
-          [key, await (fn as DomainFunction)(input, environment)] as const,
-      ),
-    )
+  const dfsWithKey = Object.entries(fns).map(([key, df]) =>
+    map(df, (result) => Object.fromEntries([[key, result]])),
```

Comment:
I think this is clear - as in: it is explicit it is an object with only one property 
```suggestion
    map(df, (result) => ({ [key]: result })),
```